### PR TITLE
vtk: darwin compatibility

### PR DESCRIPTION
**Pull request by acowley**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [x] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by LnL7 — APPROVED** [review]
what's the issue with the tests?

File: pkgs/development/libraries/vtk/default.nix [hunk 1]
```diff
@@ -30,7 +39,18 @@ stdenv.mkDerivation rec {
...
   cmakeFlags = [ "-DCMAKE_C_FLAGS=-fPIC" "-DCMAKE_CXX_FLAGS=-fPIC" ]
-    ++ optional (qtLib != null) [ "-DVTK_USE_QT:BOOL=ON" ];
+    ++ optional (qtLib != null) [ "-DVTK_USE_QT:BOOL=ON" ]
+    ++ optional stdenv.isDarwin [ "-DBUILD_TESTING:BOOL=OFF"
+                                  "-DCMAKE_OSX_SYSROOT="
+                                  "-DCMAKE_OSX_DEPLOYMENT_TARGET=" ];
+
+  doCheck = !stdenv.isDarwin;
+
+  patchPhase = stdenv.lib.optionalString stdenv.isDarwin ''
```
Comment (joachifm) [inline pkgs/development/libraries/vtk/default.nix]: Please consider using `postPatch`, overriding the standard patch phase breaks e.g., `patches` (not used here, but still).

**Comment by DamienCassou** [comment]
I know nothing about darwin, sorry.

File: pkgs/development/libraries/vtk/default.nix (hunk 1 above)
Comment (acowley) [inline pkgs/development/libraries/vtk/default.nix]: Thanks, that's very good advice.

**Comment by acowley** [comment]
Thank you, @joachifm!

**Pull request merged**